src/atopile/front_end.py

```diff
@@ -1049,43 +1124,36 @@ def visitAssert_stmt(self, ctx: ap.Assert_stmtContext) -> KeyOptMap:
         return KeyOptMap.empty()
 
     def visitComparison(self, ctx: ap.ComparisonContext) -> KeyOptMap:
-        _visited = FuncDict[ParserRuleContext, fab_param.Parameter]()
-
-        def _visit(ctx: ParserRuleContext) -> fab_param.Parameter:
-            if ctx in _visited:
-                return _visited[ctx]
-            param = self.visit(ctx)
-            _visited[ctx] = param
-            return param
-
-        params = []
-        for lh_ctx, rh_ctx in itertools.pairwise(
-            itertools.chain([ctx.arithmetic_expression()], ctx.compare_op_pair())
-        ):
-            lh = _visit(lh_ctx.arithmetic_expression())
-            rh = _visit(rh_ctx.arithmetic_expression())
-
-            # HACK: this is a cheap way to get the operator string
-            operator_str: str = rh_ctx.getChild(0).getText()
-            match operator_str:
+        exprs = [
```

Comment:
Assertions were failing because the `lh` and `rh` sides were variable. This approach guarantees operands + an operator better